src/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC } from 'react';
+import { DrawerContainer } from './drawerStyles';
+import theme from 'src/styles/theme';
+import { ThemeProvider } from '@emotion/react';
+import Button from 'src/Button/Button';
+import { ReactComponent as CloseIcon } from './assets/closeIcon.svg';
+
+export interface DrawerProps {
+    /**
+     * This is the initial display state of the component. When true, the drawer will be displayed.
+     * When false, the drawer will be hidden.
+     */
+    open: boolean;
+    /**
+     * This is an action to run when the drawer is closed.
+     */
+    onClose(): void;
+    /**
+     * This is the width for the drawer.
+     */
+    width?: string | number;
+    /**
+     * These are the classes that would be applied to drawer.
+     */
+    className?: string;
+    /**
+     * These are the classes for the toolbar.
+     */
+    toolbarClasses?: string;
+}
+
+/**
+ * Drawable Component
+ *
+ * The Drawable component can be used to open the draw and show the contents of the drawer.
+ * The component takes drawer props. The `open` prop
+ * controls the initial state of the component. Once initialized, the state is managed within this component.
```

Comment:
The state is managed by the user (each time they pass a new value for the `open` prop), not within this component.